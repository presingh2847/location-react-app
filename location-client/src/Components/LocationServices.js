export async function getLocation(inputIp) {
  try {
    const requestOptions = {
      method: "GET",
      headers: { "Content-Type": "application/json" },
    };
    const response = await fetch(`/api/geoData?ip=${inputIp}`, requestOptions);
    if (!response.ok) {
      const message = "Address was not found";
      return Promise.reject(message);
    }
    return await response.json();
  } catch (error) {
    return Promise.reject("Something bad happened, try again");
  }
}
