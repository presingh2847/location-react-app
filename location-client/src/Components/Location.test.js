//Test file for location component
import React from "react";
import MyLocation from "./Location";
import snapshotRenderer from "react-test-renderer";
import * as LocationServices from "./LocationServices";
import {
  render,
  cleanup,
  fireEvent,
  screen,
  waitFor,
} from "@testing-library/react"; //For inspection renderingimport snapshotRenderer from "react-test-renderer"; //For snapshot renderingimport thunkMiddleware from "redux-thunk";import BreachActions from "./BreachActions";import { mount } from "enzyme"; //For shallow DOM creation and inspection

let initState = {};
const MyLocationTestComponent = () => {
  return <MyLocation />;
};
describe("My Location Test Component", () => {
  beforeEach(() => {
    LocationServices.getLocation = jest.fn(() =>
      Promise.resolve({
        location: {
          latitude: 123,
          longitude: 123,
        },
      })
    );
    initState = {
      inputVal: null,
      geoInfo: null,
      showError: false,
    };
  });
  afterEach(() => {
    cleanup();
  });
  it("renders location component snapshot", () => {
    const renderedSnapshot = snapshotRenderer.create(
      <MyLocationTestComponent />
    );
    expect(renderedSnapshot.toJSON()).toMatchSnapshot();
  });
  it("Displays error when inavlid ip address is entered", () => {
    render(<MyLocationTestComponent />);
    const userInput = screen.getByRole("textbox");
    fireEvent.change(userInput, {
      target: { value: "11111111111" },
    });
    const findButton = screen.getByRole("button");
    fireEvent.click(findButton);
    const errorMessage = screen.getByTestId("invalidError");
    expect(errorMessage).toBeDefined();
  });
  it("Mock service function is called when user clicks find location with valid input", async () => {
    render(<MyLocationTestComponent />);
    const userInput = screen.getByRole("textbox");
    fireEvent.change(userInput, {
      target: { value: "204.120.0.15" },
    });
    const findButton = screen.getByRole("button");
    fireEvent.click(findButton);
    const listNode = await waitFor(() => screen.getByTestId("locationBlock"));
    expect(LocationServices.getLocation).toHaveBeenCalled();
    expect(listNode).toBeDefined();
  });
});
