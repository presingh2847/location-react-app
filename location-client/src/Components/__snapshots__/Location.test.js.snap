// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Location Test Component renders location component snapshot 1`] = `
<div
  className="location"
>
  <div
    className="location__Container"
  >
    <svg
      fill="currentColor"
      height={70}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 12 16"
      width={70}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 0C2.69 0 0 2.5 0 5.5 0 10.02 6 16 6 16s6-5.98 6-10.5C12 2.5 9.31 0 6 0zm0 14.55C4.14 12.52 1 8.44 1 5.5 1 3.02 3.25 1 6 1c1.34 0 2.61.48 3.56 1.36.92.86 1.44 1.97 1.44 3.14 0 2.94-3.14 7.02-5 9.05zM8 5.5c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"
        fillRule="evenodd"
      />
    </svg>
    <h1
      className="location__Header"
    >
      What is my location?
    </h1>
    <form
      className="location__form"
      onSubmit={[Function]}
    >
      <label>
        Enter your IP
        :
        <input
          autoComplete="off"
          className="formInput"
          name="ip"
          onChange={[Function]}
          type="text"
        />
      </label>
      <button
        className="btn btn-info btn-md"
      >
        Find Location
         
      </button>
    </form>
  </div>
</div>
`;
